<launch>
  <!-- Arguments -->
  <arg name="node_name" default="inverse_transformer_bagger"/>

  <arg name="new_poses_frame_id" default="odom"/>
  <arg name="robot_odom_frame_id" default="odom"/>
  <arg name="robot_frame_id" default="robot"/>

  <arg name="poses" default="/fused_poses"/>
  <arg name="poses_odom" default="/fused_poses_odom"/>

  <arg name="input_bag_file" default="/home/said/fxpbags/tracking/trackpathrvr1.bag"/>

  <!-- Node launching -->
  <node pkg="mercator_sensor_fusion_ros_pkg" type="inverse_transformer_bagger" name="$(arg node_name)" output="screen">
    <!-- Parameters-->
    <param name="new_poses_frame_id" type="string" value="$(arg new_poses_frame_id)" />
    <param name="robot_odom_frame_id" type="string" value="$(arg robot_odom_frame_id)" />
    <param name="robot_frame_id" type="string" value="$(arg robot_frame_id)" />
    <param name="input_bag_file" type="string" value="$(arg input_bag_file)" />
    <!-- Topic remappings -->
    <remap from="poses" to="$(arg poses)" />
    <remap from="poses_odom" to="$(arg poses_odom)" />
  </node>
</launch>
